spec/integration/posts_spec.rb

```diff
@@ -0,0 +1,127 @@
+
+RSpec.describe User, type: :feature do
+  let(:person1) do
+    User.new(name: 'Bilal', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'I am full stack developer',
+             posts_counter: 0)
+  end
+
+  let(:person2) do
+    User.new(name: 'Ashna', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'I am full stack developer',
+             posts_counter: 0)
+  end
+
+  let(:first_post) do
+    Post.new(author_id: person1.id, title: 'Hello this is the first post', text: 'Most important description',
+             likes_counter: 0, comments_counter: 0)
+  end
+
+  let(:second_post) do
+    Post.new(author_id: person1.id, title: 'Hello-2 this is the second post', text: 'Most important description-2',
+             likes_counter: 0, comments_counter: 0)
+  end
+
+  let(:first_comment) do
+    Comment.new(author_id: person2.id, post: first_post, text: 'This is the first comment on first post')
+  end
+
+  let(:second_comment) do
+    Comment.new(author_id: person2.id, post: first_post, text: 'This is the second comment on first post')
+  end
+
+  let(:third_comment) do
+    Comment.new(author_id: person2.id, post: first_post, text: 'This is the third comment on first post')
+  end
+
+  before do
+    person1.save!
+    person2.save!
+    first_post.save!
+    second_post.save!
+    first_comment.save!
+    second_comment.save!
+    third_comment.save!
+  end
+  it "Will check if the user's profile picture displays" do
+    visit "/users/#{person1.id}/posts"
+    expect(page).to have_css("img##{person1.id}")
+  end
+
+  it 'Will check if the username of user displays' do
+    visit "/users/#{person1.id}/posts"
+    expect(page).to have_content(person1.name)
+  end
+
+  it 'Will check the number of posts visible for user' do
+    visit "/users/#{person1.id}/posts"
+    expect(page).to have_content(person1.posts_counter)
+  end
+
+  it 'Will check if the posts title displays' do
+    visit "/users/#{person1.id}/posts"
+    expect(page).to have_content(first_post.title)
+  end
+
+  it 'Will check if the post body displays' do
+    visit "/users/#{person1.id}/posts"
+    expect(page).to have_content(first_post.text)
+  end
+
+  it 'Will check if the first comment displays' do
+    visit "/users/#{person1.id}/posts"
+    id = first_comment.id
+    first_comment = find("#comment-#{id}", wait: 10)
+    expect(page).to have_content(first_comment.text)
+  end
+
+  it 'Will check if the comments counter visible' do
+    visit "/users/#{person1.id}/posts"
+    expect(page).to have_content(first_post.comments_counter)
+  end
+
+  it 'Will check if the likes counter visible' do
+    visit "/users/#{person1.id}/posts"
+    expect(page).to have_content(first_post.likes_counter)
+  end
+
+  it "Will check if I click on post it redirects to post's show page" do
+    visit "/users/#{person1.id}/posts"
+    second_post_link = find('.post_item', match: :first)
+    second_post_link.click
+    expect(page).to have_content(second_post.title)
```

Comment:
- [ ] I ran all your test three times and found that one test keeps failing. See the message in the image below 👇🏾 
The failure I found is from line 91 in the post_spec.rb file. Please fix this and re-submit. 


![image](https://github.com/bilalrajput09/Blog_app_rails/assets/84156714/30984b81-bfe3-473d-99e8-2d67869cf8dc)